src/proof_rules/ost.rs

```diff
@@ -90,6 +92,16 @@ impl Encoding for OSTAnnotation {
         resolve.visit_expr(post)
     }
 
+    fn check_calculus(&self, calculus: &Calculus, direction: Direction) -> Result<(), ()> {
+        if let CalculusType::WP = calculus.calculus_type {
```

Comment:
Why not `calculus.calculus_type == CalculusType::Wp && direction == Direction::Down`?